cmd/go-yaml/node.go

```diff
@@ -87,24 +135,199 @@ func formatStyle(s yaml.Style) string {
 		return "Flow"
+	case 0:
+		// Plain style - only show if profuse
+		if profuse {
+			return "Plain"
+		}
 	}
 	return ""
 }
 
+// formatStyleName converts a YAML node style into a lowercase style name.
+// Always returns a style name (defaults to "plain" for style 0).
+func formatStyleName(s yaml.Style) string {
+	// Remove tagged style bit for checking base style
+	baseStyle := s &^ yaml.TaggedStyle
+
+	switch baseStyle {
+	case yaml.DoubleQuotedStyle:
+		return "double"
+	case yaml.SingleQuotedStyle:
+		return "single"
+	case yaml.LiteralStyle:
+		return "literal"
+	case yaml.FoldedStyle:
+		return "folded"
+	case yaml.FlowStyle:
+		return "flow"
+	default:
+		return "plain"
+	}
+}
+
 // formatTag converts a YAML tag string to its string representation.
-func formatTag(tag string, style yaml.Style) string {
+func formatTag(tag string, style yaml.Style, profuse bool) string {
 	// Check if the tag was explicit in the input
 	tagWasExplicit := style&yaml.TaggedStyle != 0
 
-	// Show !!str only if it was explicit in the input
+	// In profuse mode, always show tag
+	if profuse {
+		return tag
+	}
+
+	// Default YAML tags - only show if they were explicit in the input
 	switch tag {
-	case "!!str", "!!map", "!!seq":
+	case "!!str", "!!map", "!!seq", "!!int", "!!float", "!!bool", "!!null":
 		if tagWasExplicit {
 			return tag
 		}
 		return ""
 	}
 
-	// Show all other tags
+	// Show all other tags (custom tags)
 	return tag
 }
+
+// FormatNodeCompact converts a YAML node into a compact representation.
+// Document nodes return their content directly.
+// Mapping/Sequence nodes use lowercase keys: "mapping:", "sequence:".
+// Scalar nodes use style as key: "plain:", "double:", etc.
+func FormatNodeCompact(n yaml.Node) interface{} {
+	switch n.Kind {
+	case yaml.DocumentNode:
+		// Check if document has properties that need to be preserved
+		hasProperties := n.Anchor != "" || n.HeadComment != "" || n.LineComment != "" || n.FootComment != ""
+		if tag := formatTag(n.Tag, n.Style, false); tag != "" && tag != "!!str" {
+			hasProperties = true
+		}
+
+		// If document has no properties, return content directly (unwrap)
+		if !hasProperties {
+			if len(n.Content) > 0 {
+				return FormatNodeCompact(*n.Content[0])
+			}
+			return nil
+		}
+
+		// Document has properties - create a result map with ordered keys
+		result := MapSlice{}
+
+		// Add optional fields in order: tag, anchor, comments
+		if tag := formatTag(n.Tag, n.Style, false); tag != "" && tag != "!!str" {
+			result = append(result, MapItem{Key: "tag", Value: tag})
+		}
+		if n.Anchor != "" {
+			result = append(result, MapItem{Key: "anchor", Value: n.Anchor})
+		}
+		if n.HeadComment != "" {
+			result = append(result, MapItem{Key: "head", Value: n.HeadComment})
+		}
+		if n.LineComment != "" {
+			result = append(result, MapItem{Key: "line", Value: n.LineComment})
+		}
+		if n.FootComment != "" {
+			result = append(result, MapItem{Key: "foot", Value: n.FootComment})
```

Comment:
!!str, tag, head, foot, …

All of them could be constants

It would avoid possible errors such as typo like "stt", "hesd"

Reply:
for @SVilgelm